jdbc-source/src/main/java/org/springframework/cloud/stream/module/jdbc/source/JdbcSourceConfiguration.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.jdbc.source;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.module.MaxMessagesProperties;
+import org.springframework.cloud.stream.module.PeriodicTriggerConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.integration.core.MessageSource;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.channel.MessageChannels;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.jdbc.JdbcPollingChannelAdapter;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A module that reads data from an RDBMS using JDBC and creates a payload with the data.
+ *
+ * @author Thomas Risberg
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({JdbcSourceProperties.class, MaxMessagesProperties.class})
+@Import({PeriodicTriggerConfiguration.class})
+public class JdbcSourceConfiguration {
+
+	@Autowired
+	private JdbcSourceProperties properties;
+
+	@Autowired
+	DataSource dataSource;
+
+	@Autowired
+	PlatformTransactionManager transactionManager;
+
+	@Autowired
+	@Bindings(JdbcSourceConfiguration.class)
+	Source source;
+
+	@Bean
+	TaskExecutor taskExecutor() {
+		return new ThreadPoolTaskExecutor();
+	}
+
+	@Bean
+	@ConditionalOnProperty(value = "split", havingValue = "true", matchIfMissing = true)
+	public IntegrationFlow splitFlow() {
+		return IntegrationFlows.from("splitter")
+				.split("payload")
+				.channel(MessageChannels.executor(this.taskExecutor()))
+				.channel(source.output())
```

Comment:
I don't think that we should worry about the `output` performance and parallelism.
The `source.output()` is pipe for the `Binder` - Redis, Kafka, Rabbit etc.

I mean that this `MessageChannels.executor()` is overhead.
And from here we even can get rid of this `IntegrationFlow` for the `split` option.